openml_OS/views/pages/api_new/v1/xsd/openml.study.upload.xsd

```diff
@@ -0,0 +1,82 @@
+<!-- Schema for dataset description. Noe that it is used both for 
+     uploading and downloading a dataset. Some fields, such as ID, 
+     upload_date, are ignored during uploading, since these are to be 
+     generated by the server. -->
+
+<xs:schema targetNamespace="http://openml.org/openml"
+    xmlns:oml="http://openml.org/openml"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+    elementFormDefault="qualified">
+<xs:element name="study" type="oml:study"/>
+
+<xs:complexType name="study">
+  <xs:sequence>
+    <xs:element name="id" minOccurs="0" type="xs:int"/>  <!-- ID autogenerated by the server. Integer. Added by the server. Ignored when uploaded. -->
+    <xs:element name="alias" minOccurs="0" type="oml:system_string32"/> <!-- requested name to obtain. URL friendly --> 
+    <xs:element name="main_entity_type" type="oml:entity_type"/> <!-- whether it is a collection of runs (study) or collection of tasks (benchmark suite) --> 
+    <xs:element name="benchmark_suite" minOccurs="0" type="xs:int"/>
+    <xs:element name="name" type="oml:basic_latin256"/> <!-- requested name to obtain. URL friendly --> 
+    <xs:element name="description" minOccurs="0" type="oml:basic_latin4096"/>
+    <xs:element name="creation_date" minOccurs="0" type="xs:dateTime"/>  <!-- The date-time, generated by server, on which the dataset was uploaded. Format YYYY-mm-dd hh:MM:SS. Added by the server. Ignored when uploaded manually. -->
+    <xs:element name="creator" minOccurs="0" type="xs:int"/> <!-- ID added by the server. Integer. Added by the server. Ignored when uploaded. -->
+    <xs:element name="data" minOccurs="0" type="oml:data"/>
+    <xs:element name="tasks" minOccurs="0" type="oml:tasks"/>
+    <xs:element name="flows" minOccurs="0" type="oml:flows"/>
+    <xs:element name="setups" minOccurs="0" type="oml:setups"/>
+    <xs:element name="runs" minOccurs="0" type="oml:runs"/>
+  </xs:sequence>
+</xs:complexType>
+
+<xs:simpleType name="entity_type">
+  <xs:restriction base="xs:string">
+    <xs:pattern value="(run|task)" />
+  </xs:restriction>
+</xs:simpleType>
+<xs:simpleType name="system_string32"> <!-- Subset on xs:string. Highly restricted form of string. URL-friendly -->
+  <xs:restriction base="xs:string">
+    <xs:pattern value="([a-zA-Z0-9_\-\.])+" />
+    <xs:maxLength value="32" />
+    <xs:minLength value="1" />
+  </xs:restriction>
+</xs:simpleType>
+<xs:simpleType name="basic_latin256">
+  <xs:restriction base="xs:string">
+    <xs:pattern value="\p{IsBasicLatin}*"/>
+    <xs:maxLength value="256" />
+    <xs:minLength value="1" />
+  </xs:restriction>
+</xs:simpleType>
+<xs:simpleType name="basic_latin4096">
+  <xs:restriction base="xs:string">
+    <xs:pattern value="\p{IsBasicLatin}*"/>
+    <xs:maxLength value="4096" />
+    <xs:minLength value="1" />
+  </xs:restriction>
+</xs:simpleType>
+
+<xs:complexType name="data">
```

Comment:
Isn't it overkill to have separate XSD types for what are basically just IDs? Why not let all of these be normal lists of integers?

Reply:
it's because of 

```
<entity type>
  <entity id></entity id>
<entity type>
```